docs/dev/shims.md

```diff
@@ -17,7 +17,7 @@ common code, maximize reuse, and minimize logic duplication.
 This is achieved by using a ServiceProvider pattern. All Shims implement the same API,
 the suitable Shim implementation is loaded after detecting the current Spark build version
 attempting to instantiate our plugin. We use the
-[ShimLoader](https://github.com/NVIDIA/spark-rapids/blob/main/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala)
+[ShimLoader](../../sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala)
```

Comment:
There's no source code in the gh-pages branch, so this change likely breaks the Jekyll site when published.  Kind of a minor comment since I don't think most devs are looking at the gh-pages site in practice.  I'm wondering if we should stop publishing there. :thinking:

Reply:
I did not know about the separate branch for Jekyll. 

> I'm wondering if we should stop publishing there. :thinking: 

Do you meaning moving `docs/dev/shim*.md` to a location not picked up by Jekyll? This makes sense to me.

Reply:
I meant removing docs/dev from the gh-pages branch.  I see gh-pages more for end-users than developers who are going to have the repo checked out and browsing code/github markedown directly for dev documentation.

Reply:
ok, I'll undo the link changes in this PR.  Switching to relative paths can be done after docs/dev is removed from gh-pages